_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1053,6 +1054,14 @@ private function _process_onboarding( $data ) {
 			}
 		}
 
+		if ( isset( $data['businessAddress'] ) ) {
+			$handled_business_address = self::handle_business_address( $data['businessAddress'] );
+			if ( is_wp_error( $handled_business_address ) ) {
+				$error[] = 'BusinessAddress';
+			}
+			update_option( 'jpo_business_address', $data['businessAddress'] );
```

Comment:
Right now we update the business address regardless of whether it worked properly or not. We should move this to the bottom `handle_business_address()`, right after the insert/update operation has completed successfully.